src/ConcordiumNetSdk/Client.cs

```diff
@@ -0,0 +1,818 @@
+﻿using Concordium.V2;
+using Grpc.Core;
+using Grpc.Net.Client;
+
+namespace ConcordiumNetSdk;
+
+/// <summary>
+/// A client for interacting with the GRPC V2 API exposed by Concordium nodes.
+/// </summary>
+public class Client : IDisposable
+{
+    /// <summary>
+    /// The "internal" client instance generated from the V2 API protocol buffer definition.
+    /// </summary>
+    private readonly Queries.QueriesClient _client;
+
+    /// <summary>
+    /// The channel on which the client communicates with the Concodium node.
+    /// </summary>
+    private readonly GrpcChannel _grpcChannel;
+
+    /// <summary>
+    /// The configuration to use with the client.
+    /// </summary>
+    private readonly ClientConfiguration _config;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Client"/> class.
+    /// Uses the <see cref="ClientConfiguration.DefaultConfiguration"/> for the client settings.
+    /// </summary>
+    /// <param name="url">URL of a resource where the V2 API is served.</param>
+    /// <param name="port">Port of the resource where the V2 API is served.</param>
+    public Client(string url, UInt16 port)
+        : this(url, port, ClientConfiguration.DefaultConfiguration) { }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Client"/> class.
+    /// </summary>
+    /// <param name="url">URL of a resource where the V2 API is served.</param>
+    /// <param name="port">Port of the resource where the V2 API is served.</param>
+    /// <param name="timeout">The request timeout in seconds.</param>
+    /// <param name="secure">Flag indicating whether the client must use a secure connection.</param>
+    public Client(string url, UInt16 port, ulong timeout, bool secure)
+        : this(url, port, new ClientConfiguration(timeout, secure)) { }
```

Comment:
It wuld be better to have a ClientBuilder instead of these multitude of constructors.
 
See https://docs.rs/reqwest/0.11.16/reqwest/struct.ClientBuilder.html for the pattern.

Reply:
Ok.